confmap/provider.go

```diff
@@ -8,6 +8,10 @@ import (
 	"fmt"
 )
 
+// ProviderSettings are the settings to initialize a Provider.
+// Any Provider should take this as a parameter in its constructor.
+type ProviderSettings struct{}
```

Comment:
Should we have "global" settings or each provider defines it's own settings? I would say better to have each provider with it's settings. The argument that we can have a property added here and passed everywhere is not strong since we still need to change each provider constructor to use it, so because of that I would simply go with a `Settings` struct in each package and deal with possible duplicate definition of properties.

Pros:
 * Can involve independently;

Cons:
 * Duplicate properties;

Reply:
Having independent settings would make harder fixing #4759: the builder would need to know what settings to provide for each component. With a single shared settings struct that we control we don't have that problem. In general, we have not allowed settings to vary across components (e.g. for exporters, connectors, extensions, receivers, processors... they each have a different but shared set of settings to initialize components), why should we do it in this case? Do you have specific ideas as to what settings could be independent across providers?

Reply:
You are correct, I think we should stick with your proposal.